tests/test_solve.py

```diff
@@ -84,7 +86,7 @@ def test_solve_custom_params(rc208):
     expected by checking how solutions are accepted.
     """
 
-    def monotonically_decreasing(arr) -> np.bool:
+    def monotonically_decreasing(arr) -> np.bool_:
```

Comment:
`np.bool` and `np.bool_` are aliases, right? Is there a reason to prefer `np.bool_`?

Reply:
I'm a bit confused myself here. I got this error on numpy v1.26:
```
E           AttributeError: module 'numpy' has no attribute 'bool'.
````

Turns out that: 
- `np.bool` got deprecated in [v1.20](https://numpy.org/doc/2.0/release/1.20.0-notes.html#using-the-aliases-of-builtin-types-like-np-int-is-deprecated)
- Removed in [v1.24](https://numpy.org/doc/2.0/release/1.24.0-notes.html), and
- Added back again as dtype in [v2.0](https://numpy.org/devdocs/release/2.0.0-notes.html#changes).

So there's a little gap in versions where `np.bool` won't work, so I think it's best to keep it as `np.bool_` for now.